scanpy/get.py

```diff
@@ -194,6 +204,23 @@ def obs_df(
             f" {gene_error}."
         )
 
+    # check that adata.obs.index is unique
+    orig_index_map = None
+    if not adata.obs.index.is_unique:
+        # non unique obs index is problematic in the `join` operation used in the next
+        # lines which will erroneously expand the length of the obs adata frame.
+        # To solve the problem, a unique index is temporarily created which is
+        # afterwards replaced by the original index
+        logg.warning(
+            'Warning: the adata.obs index is not unique. This can be problematic for '
+            'operations with adata.obs '
+        )
```

Comment:
Use actual `warnings`, they are easier for users to deal with.

Reply:
done